libraries/extractor/src/main/java/androidx/media3/extractor/DtsUtil.java

```diff
@@ -199,8 +334,262 @@ public static int getDtsFrameSize(byte[] data) {
     return uses14BitPerWord ? fsize * 16 / 14 : fsize;
   }
 
+  /**
+   * Returns the DTS format given {@code data} containing the DTS-HD frame(containing only Extension
+   * Sub-stream) according to ETSI TS 102 114 V1.6.1 (2019-08), Section 7.4/7.5.
+   *
+   * @param frame The DTS-HD frame(containing only Extension Sub-stream) to parse.
+   * @return The DTS format parsed from data in the header.
+   */
+  public static DtsFormatInfo parseDtsHdFormat(byte[] frame) {
+    ParsableBitArray frameBits = getNormalizedFrameHeader(frame);
+    frameBits.skipBits(32 + 8); // SYNC, userDefinedBits
+    int extSSIndex = frameBits.readBits(2);
+    int headerSizeType = frameBits.readBits(1);
+    int nBits4Header, nBits4ExSSFsize;
+    if (headerSizeType == 0){
+      nBits4Header = 8;
+      nBits4ExSSFsize = 16;
+    }
+    else{
+      nBits4Header = 12;
+      nBits4ExSSFsize = 20;
+    }
+    frameBits.skipBits(nBits4Header); // Extension substream header size
+    int extFrameSize = frameBits.readBits(nBits4ExSSFsize) + 1; // Extension substream frame size
+
+    int numAudioPresent = 1, numAssets = 1;
+    int nRefClockCode = 0, nExSSFrameDurationCode = 0;
+    int staticFieldsPresent = frameBits.readBits(1);
+    if (staticFieldsPresent == 1) {
+      nRefClockCode = frameBits.readBits(2);
+      nExSSFrameDurationCode = 512 * (frameBits.readBits(3) + 1);
+      if (frameBits.readBits(1) == 1) {
+        frameBits.skipBits(32 + 4);
+      }
+
+      int[] nuActiveExSSMask = new int[8];
+      frameBits.skipBits(3 + 3); // numAudioPresent, numAssets
```

Comment:
Why do we skip the bits for `numAudioPresent` and `numAssests` here? Based on the spec it seems that both `numAudioPresent` and `numAssests` should be 1 only if static fields are not present but in the current code it is always 1 irrespective of whether static fields are present or not.

Reply:
Streaming encoders won't encode streams with multiple audio presentations and assets.
In order to make the parsing logic less complex, I have hardcoded those values as one.